Source/Libraries/openXDA.Nodes/Types/SSAMS/SSAMSNode.cs

```diff
@@ -0,0 +1,149 @@
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  02/08/2021 - Stephen C. Wills
+//       Generated original version of source code.
+//
+//******************************************************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Globalization;
+using System.Web.Http;
+using System.Web.Http.Controllers;
+using GSF;
+using GSF.Configuration;
+using GSF.Data;
+using GSF.Data.Model;
+using GSF.Threading;
+using log4net;
+using openXDA.Configuration;
+using openXDA.Model;
+
+namespace openXDA.Nodes.Types.SSAMS
+{
+    public class SSAMSNode : NodeBase
+    {
+        #region [ Members ]
+
+        // Nested Types
+        private class Settings
+        {
+            public Settings(Action<object> configure) =>
+                configure(this);
+
+            [Category]
+            [SettingName(SSAMSSection.CategoryName)]
+            public SSAMSSection SSAMSSettings { get; } = new SSAMSSection();
+        }
+
+        private class SSAMSWebController : ApiController
+        {
+            private SSAMSNode Node { get; }
+
+            public SSAMSWebController(SSAMSNode node) =>
+                Node = node;
+        }
+
+        #endregion
+
+        #region [ Constructors ]
+
+        public SSAMSNode(Host host, Node definition, NodeType type)
+            : base(host, definition, type)
+        {
+            Action<object> configurator = GetConfigurator();
+            ScheduleDBSignal(configurator);
+        }
+
+        #endregion
+
+        #region [ Properties ]
+        private Func<AdoDataConnection> ConnectionFactory { get; set; }
+
+        #endregion
+
+        #region [ Methods ]
+
+        public override IHttpController CreateWebController() =>
+            new SSAMSWebController(this);
+
+        protected override void OnReconfigure(Action<object> configurator)
+        {
+            ScheduleDBSignal(configurator);
+        }
+
+        private void ScheduleDBSignal(Action<object> configurator)
+        {
+            Settings settings = new Settings(configurator);
+            string name = nameof(DatabaseOperation);
+            string schedule = settings.SSAMSSettings.Schedule;
+            ConnectionFactory = () => new AdoDataConnection(settings.SSAMSSettings.ConnectionString, settings.SSAMSSettings.DataProviderString);
+            Host.RegisterScheduledProcess(this, DatabaseOperation, name, schedule);
+        }
+
+        private void DatabaseOperation()
+        {
+            Action<object> configurator = GetConfigurator();
+            Settings settings = new Settings(configurator);
+            if (settings.SSAMSSettings.DatabaseCommand is null)
+                return;
+            using (AdoDataConnection connection = ConnectionFactory())
+            {
+                List<object> parameters = new List<object>();
```

Comment:
This is on me, but if we use `{TimeStamp}` in the Settings we will need to do a replacement for that here. I would recommend just replacing it with the value of `DateTime.UTCNow`